packages/blockchain-api/src/currencyConversion/CurrencyConversionAPI.test.ts

```diff
@@ -0,0 +1,134 @@
+import { InMemoryLRUCache } from 'apollo-server-caching'
+import BigNumber from 'bignumber.js'
+import CurrencyConversionAPI from './CurrencyConversionAPI'
+import ExchangeRateAPI from './ExchangeRateAPI'
+import GoldExchangeRateAPI from './GoldExchangeRateAPI'
+
+jest.mock('./ExchangeRateAPI')
+jest.mock('./GoldExchangeRateAPI')
+
+const mockDefaultGetExchangeRate = ExchangeRateAPI.prototype.getExchangeRate as jest.Mock
+mockDefaultGetExchangeRate.mockResolvedValue(new BigNumber(20))
+
+const mockGoldGetExchangeRate = GoldExchangeRateAPI.prototype.getExchangeRate as jest.Mock
+mockGoldGetExchangeRate.mockResolvedValue(new BigNumber(10))
+
+describe('CurrencyConversionAPI', () => {
+  let currencyConversionAPI: CurrencyConversionAPI
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    currencyConversionAPI = new CurrencyConversionAPI()
+    currencyConversionAPI.initialize({ context: {}, cache: new InMemoryLRUCache() })
+  })
+
+  it('should retrieve rate for cGLD/cUSD', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'cGLD',
+      currencyCode: 'cUSD',
+    })
+    expect(result).toEqual(new BigNumber(10))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(0)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for cUSD/cGLD', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'cUSD',
+      currencyCode: 'cGLD',
+    })
+    expect(result).toEqual(new BigNumber(10))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(0)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for cGLD/USD', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'cGLD',
+      currencyCode: 'USD',
+    })
+    expect(result).toEqual(new BigNumber(10))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(0)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for USD/cGLD', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'USD',
+      currencyCode: 'cGLD',
+    })
+    expect(result).toEqual(new BigNumber(10))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(0)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for cGLD/MXN', async () => {
+    const result = await currencyConversionAPI.getExchangeRate({
+      sourceCurrencyCode: 'cGLD',
+      currencyCode: 'MXN',
+    })
+    expect(result).toEqual(new BigNumber(200))
+    expect(mockDefaultGetExchangeRate).toHaveBeenCalledTimes(1)
+    expect(mockGoldGetExchangeRate).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retrieve rate for MXN/cGLD', async () => {
```

Comment:
Does this test cover any additional logic from the `MXN/cGLD` one given the mocks?

Reply:
They look really similar indeed but the idea was to cover both Local to cGLD and cGLD to local.